modules/tinymce/src/core/main/ts/textpatterns/core/BlockPattern.ts

```diff
@@ -55,10 +55,18 @@ const applyPattern = (editor: Editor, match: BlockPatternMatch): boolean => {
   return true;
 };
 
+const sortPatterns = <P extends Pattern>(patterns: P[]): P[] => Arr.sort(patterns, (a, b) => {
+  if (a.start.length === b.start.length) {
+    return 0;
+  }
+  return a.start.length > b.start.length ? -1 : 1;
+});
+
 // Finds a matching pattern to the specified text
 const findPattern = <P extends Pattern>(patterns: P[], text: string): Optional<P> => {
+  const sortedPatterns = sortPatterns(patterns);
```

Comment:
Instead of sorting them every time they're used, can't we sort them when the pattern set is created?

Reply:
Sorting when the pattern was set, was the original logic and it caused the bug that this PR is meant to fix

Reply:
The other thing to remember is that the pattern set is generated every time we attempt to find the patterns anyways due to the dynamic lookups, so moving it there would make no difference anyways.

Reply:
Also as I noted in a previous comment, doing it when we create the pattern set is actually worse for performance in the  "inline key" case since it never uses block patterns.